### PR TITLE
WIP: Add cloud testing + refactoring nominal x y replacement

### PR DESCRIPTION
**Pull request by jbusecke**
Towards fixing #30 and #34

### CODE REVIEW

**Comment by jbusecke** [comment]
@andersy005: This is the PR I was talking about earlier. It seems like the circleci is running but its so slow that it wont even finish a fraction of the test, so that is not feasible at the moment. Could you think of a way to speed the reading with intake-esm or get more bandwidth?

For travis it really seems like they dont allow access from within their testing env...see this cryptic error [here])(https://travis-ci.com/github/jbusecke/cmip6_preprocessing/jobs/314580159).
If you got any input how to speed this up, or who to contact with this, I would tremendously appreciate the help. I think this testing is awesome (even though its slow) to make sure that changes to cmip6_preprocessing are working on input in the cloud.

File: cmip6_preprocessing/tests/test_preprocessing_cloud.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+@pytest.mark.parametrize("source_id", all_models())
+def test_lat_lon(col, source_id, experiment_id, grid_label, variable_id):
+    cat = col.search(
+        source_id=source_id,
+        experiment_id=experiment_id,
+        variable_id=variable_id,
+        table_id="Omon",
+        grid_label=grid_label,
+    )
+    ddict = cat.to_dataset_dict(
```
Comment (andersy005) [inline cmip6_preprocessing/tests/test_preprocessing_cloud.py:55]: Try passing `storage_options` with `token=anon` to `to_dataset_dict()`

```python
ddict = cat.to_dataset_dict(zarr_kwargs={"consolidated": True, "decode_times": False}, \
           preprocess=combined_preprocessing, storage_options={'token':'anon'})

```

**Comment by andersy005** [comment]
> For travis it really seems like they dont allow access from within their testing env...see this cryptic error [here])(travis-ci.com/github/jbusecke/cmip6_preprocessing/jobs/314580159).

I found a similar issue here: https://github.com/googleapis/google-auth-library-python/issues/287. It appears that at some point TravisCI locked access to APIs (including Google's authentication API). Unfortunately the linked issue does not provided any solution (the person who opened the issue ended up migrating to CircleCI).

File: cmip6_preprocessing/tests/test_preprocessing_cloud.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@pytest.mark.parametrize("variable_id", ["thetao"])
+@pytest.mark.parametrize("source_id", all_models())
+def test_lat_lon(col, source_id, experiment_id, grid_label, variable_id):
+    cat = col.search(
+        source_id=source_id,
+        experiment_id=experiment_id,
+        variable_id=variable_id,
+        table_id="Omon",
+        grid_label=grid_label,
+    )
```
Comment (andersy005) [inline cmip6_preprocessing/tests/test_preprocessing_cloud.py:54]: For testing purposes, do you need all `member_ids`?

**Comment by andersy005** [comment]
It appears that the upstream-dev is failing due to numpy build error. Do we need to test against the master branch of NumPy i.e. would it be okay to just test against the latest stable release of NumPy?

**Comment by andersy005** [comment]
Regarding Python 3.8 and Python 3.7 jobs, I recommend re-running the jobs with SSH and SSH-ing into the container and running one of the hanging tests:

```python
pytest -v cmip6_preprocessing/tests/test_preprocessing_cloud.py::test_lat_lon[BCC-ESM1-thetao-historical-gn] 
```

**Comment by jbusecke** [comment]
@andersy005 , sorry for the delay on this. I am afraid I am not quite sure how to SSH into the container. Is there some documentation I could follow for this? Many thanks for your help.

**Comment by andersy005** [comment]
Head over to one of the jobs URL: https://app.circleci.com/pipelines/github/jbusecke/cmip6_preprocessing/26/workflows/1679fc07-8435-4abf-8537-025457a0a94d/jobs/91/steps

and click on dropdown `Rerun` button:

Screen Shot 2020-04-28 at 9 29 42 AM

> I am afraid I am not quite sure how to SSH into the container. Is there some documentation I could follow for this?

Here' the detailed documentation page: https://circleci.com/docs/2.0/ssh-access-jobs/

File: cmip6_preprocessing/tests/test_preprocessing_cloud.py (hunk 2 above)
Comment (jbusecke) [reply cmip6_preprocessing/tests/test_preprocessing_cloud.py:54]: Thats a good point. Ill try to restrict this.

**Comment by jbusecke** [comment]
Thank you so much! Ill try that out in a bit.

**Comment by jbusecke** [comment]
I think I will give up on making these tests work with CI for this PR. It is not essential for this change, and I would like to move ahead with some other changes. I will for now check these tests only locally. @rabernat mentioned [pangeo-bot](https://github.com/pangeo-gallery/binderbot) as a possible solution. I will investigate that possibility further and come back to it. I really appreciete the help on this @andersy005.

**Comment by andersy005** [comment]
> @rabernat mentioned pangeo-bot as a possible solution. I will investigate that possibility further and come back to it.

:+1: This sounds like a more robust solution.

> I really appreciete the help on this @andersy005.

Anytime! Sorry this didn't work as we had expected.

**Pull request merged**